akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -411,6 +412,34 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
       "fail entity stream if advertised content-length doesn't match" in pending
     }
 
+    "support stream for request data" should {
+      abstract class StreamingRequestSent extends TestSetup {
+        val streamId = 0x1
+        val requestStream = TestPublisher.probe[ByteString]()
+        user.emitRequest(HttpRequest(
+          protocol = HttpProtocols.`HTTP/2.0`,
+          entity = HttpEntity(ContentTypes.`application/octet-stream`, Source.fromPublisher(requestStream))))
+        network.expectDecodedHEADERS(streamId, endStream = false)
+        requestStream.expectRequest()
+      }
```

Comment:
Kind of duplicating things in #3671 here but the actual test case is not covered by that PR from what I can see.

Reply:
Would those also be useful for the server side?

Reply:
Yes, good point. 

I thought they were but now that I look at it only `send RST_STREAM when entity data stream fails` is there, which is about the response entity only.